src/vos/evt_iter.c

```diff
@@ -109,6 +109,35 @@ evt_iter_finish(daos_handle_t ih)
 	return rc;
 }
 
+/**
+ * For anchor matching, we want to match the rectangle if it's at the same epoch and
+ * there is overlap. This ensures we see extents that may get merged into an extent
+ * we've already seen.
+ */
```

Comment:
>Let's take a simple example, assuming anchor is [2, 3)@epoch 10, the merged recx is [1, 4)@epoch 20, will this compare function return '>' or '<'?

I thought aggregation will choose lower epoch?  so if [2, 3) is merged to [1, 4), its epoch should not be higher than 10?

So let's say merging extent is [1, 4) @epoch 5,  let's see this list

[0, 1)@epoch2, [1, 4) @epoch5,  [5, 7)@epoch6,  [8, 10)@epoch8  

then after binary search of [2, 3)@epoch10  in evt_iter_probe_find(), it will return mid=1 ([1, 4)epoch5.

>then how can you match the anchor?

So it does not matter whether there is matched record in this list,  but the mid (return by evt_iter_probe_find() have to be the closest one, where it is left.